accuweather/__init__.py

```diff
@@ -146,6 +150,10 @@ async def _async_get_data(self, url: str) -> dict[str, Any]:
             data = await resp.json()
         if resp.headers["RateLimit-Remaining"].isdigit():
             self._requests_remaining = int(resp.headers["RateLimit-Remaining"])
+
+        # nasty hack to account for different data structure returned by hourly forecast API call
```

Comment:
I really don't like this. Please change return type for this method to `Any` and use something like this:
```python
        if "hourly" in url:
            return data

        return data if isinstance(data, dict) else data[0]
```